src/views/StatsView.vue

```diff
@@ -115,19 +118,39 @@ export default {
       return this.seasons.map((season) => {
         return {
           title: season.name,
-          value: season
-        }
+          value: season,
+        };
       });
-    }
+    },
+  },
+  methods: {
+    checkAndSelectSeason(seasonId) {
+      if (this.seasons.some(({ id }) => id === seasonId)) {
+        const [selectedSeason] = this.seasons.filter(({ id }) => id === seasonId);
+        this.selectedSeason = selectedSeason;
+      } else {
+        const [selectedSeason] = this.seasons;
+        this.selectedSeason = selectedSeason;
+      }
```

Comment:
Might be better to fast fail here so it's easier to understand, but @itsalaidbacklife and I argue about if/else blocks all the time so I'll let him pick.
```js
const hasSeason = this.seasons.some(({ id }) => id === seasonId);

if (hasSeason) {
  const [selectedSeason] = this.seasons.filter(({ id }) => id === seasonId);
  this.selectedSeason = selectedSeason;
  return;
}

const [selectedSeason] = this.seasons;
this.selectedSeason = selectedSeason;
```

There's also probably an opportunity to not need to both do some and filter via reducer, but I don't think we need to needle on performance here.

Reply:
I think we can get rid of some here, and just filter the result, and then just check for length